src/Simplex/Chat.hs

```diff
@@ -3510,20 +3510,37 @@ processAgentMessageConn vr user@User {userId} corrId agentConnId agentMessage =
       CON _ -> Just ConnReady
       _ -> Nothing
 
+    processCONFpqSupport :: Connection -> PQSupport -> m Connection
+    processCONFpqSupport conn@Connection {connId, pqSupport} pqSupport' =
+      case (pqSupport, pqSupport') of
+        (PQSupportOn, PQSupportOff) -> do
+          withStore' $ \db -> updateConnSupportPQ db connId pqSupport'
+          pure (conn {pqSupport = pqSupport', pqEncryption = CR.pqSupportToEnc pqSupport'} :: Connection)
+        (PQSupportOff, PQSupportOn) -> do
+          messageWarning "processCONFpqSupport: unexpected PQSupportOn"
+          pure conn
+        _ -> pure conn
+
+    processINFOpqSupport :: Connection -> PQSupport -> m ()
+    processINFOpqSupport Connection {pqSupport} pqSupport' =
+      case (pqSupport, pqSupport') of
+        (PQSupport b, PQSupport b') | b /= b' -> do
+          messageWarning "processINFOpqSupport: unexpected pqSupport change"
+        _ -> pure ()
```

Comment:
```suggestion
      when (pqSupport /= pqSupport') $ messageWarning "processINFOpqSupport: unexpected pqSupport change"
```

emm

Reply:
this is not needed as function, can be in place

Reply:
kept as function to not bring pqSupport from Connection into scope of larger function / to not pattern match in place